name: test_accuracy
permissions: {} # No permissions by default on workflow level
on:
  pull_request:
  merge_group:
    branches:
      - master
concurrency:
  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
  cancel-in-progress: true
jobs:
  test_accuracy:
    runs-on: ubuntu-24.04
    steps:
      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
        with:
          persist-credentials: false
      - uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c # v6.0.0
        with:
          python-version-file: ".python-version"
      - name: Install uv
        uses: astral-sh/setup-uv@85856786d1ce8acfbcc2f13a5f3fbd6b938f9f41 # v7.1.2
      - name: Install dependencies
        run: |
          uv sync --locked --extra tests --extra-index-url https://download.pytorch.org/whl/cpu
      - name: Prepare test data
        run: |
          uv run python tests/accuracy/prepare_data.py -d data
      - name: Run Python Test
        run: |
          uv run pytest --data=./data tests/accuracy/test_accuracy.py
